src/DataGrid.tsx

```diff
@@ -363,11 +363,12 @@ function DataGrid<R, SR, K extends Key>(
   useLayoutEffect(() => {
     if (
       !selectedCellIsWithinSelectionBounds ||
-      selectedPosition === prevSelectedPosition.current ||
-      selectedPosition.mode === 'EDIT'
+      isSamePosition(selectedPosition, prevSelectedPosition.current)
     ) {
+      prevSelectedPosition.current = selectedPosition;
```

Comment:
Does it make sense to do this assignment if `isSamePosition(selectedPosition, prevSelectedPosition.current)`?

Reply:
`isSamePosition` only checks for `idx` and `rowIdx` and mode can be different